mesonbuild/build.py

```diff
@@ -228,14 +229,16 @@ def check_unity_compatible(self):
         # and create a single source file from each subset of the sources that
         # can be compiled with a specific compiler. Then we create one object
         # from each unified source file.
-        # If the list of sources for which we want objects is the same as the
-        # list of sources that go into each unified build, we're good.
+        #
+        # FIXME: Due to limitations in backends.determine_ext_objs() all
+        # extracted sources must have the same compiler. We also should check
+        # here which compiler is needed for each generated source, forbid
+        # generated source for now.
+        if self.genlist:
+            raise MesonException('Cannot extract objects from Unity builds '\
```

Comment:
__[Flake8]__

[E502] the backslash is redundant between brackets


<!-- Comment by [SideCI](https://sideci.com) -->